app/controllers/description_controller_helpers.rb

```diff
@@ -544,8 +544,14 @@ def check_description_edit_permission(desc, params)
       author = desc.is_author?(@user)
 
       params.delete(:source_type) unless root
-      params.delete(:source_name) unless root || ((admin || author) &&
-        (desc.source_type != :project && desc.source_type != :project))
+      unless root ||
+             ((admin || author) &&
+               # originally was
+               # (desc.source_type != :project && desc.source_type != :project))
+               # see https://www.pivotaltracker.com/story/show/174566300
+               desc.source_type != :project)
+        params.delete(:source_name)
+      end
```

Comment:
Fixes BinaryOperatorWithIdenticalOperands offense. See note and Pivotal Story referenced in source code.